app/models/common/single-submit-autocomplete.js

```diff
@@ -0,0 +1,48 @@
+const { errorPusherDefault } = require('../../lib/error-helpers')
+const { forms } = require('../../constants/forms')
+
+/**
+ * @typedef AddOrRemoveDetails
+ * @property {string} recordType
+ * @property {string} recordTypeText
+ * @property {string} buttonText
+ * @property {string} [optionText]
+ * @property {string} [recordValue]
+ * @property {string} [backLink]
+ * @property {'add'|'remove'} action
+ */
+
+/**
+ * @param {AddOrRemoveDetails} details
+ * @param [backNav]
+ * @param [errors]
+ * @constructor
+ */
+function ViewModel (details, backNav, errors) {
+  const inputModel = {
+    label: {
+      text: details.optionText ?? `Which ${details.recordTypeText} do you want to ${details.action}?`,
+      classes: 'govuk-label--l',
+      isPageHeading: true
+    },
+    id: details.recordType,
+    name: details.recordType,
+    value: details.recordValue ?? null,
+    autocomplete: forms.preventAutocomplete,
+    attributes: { maxlength: '8' }
```

Comment:
I don;t think this value is correct. Will it not limit the user to typing max 8 chars in the field? Perhaps limit of 100 might be better, or no limit

Reply:
just fixing this in my current commit - this is for the remove flow